src/main/java/com/szeloch/jan/parkingsystemmanagement/rest/model/Vehicle.java

```diff
@@ -0,0 +1,61 @@
+package com.szeloch.jan.parkingsystemmanagement.rest.model;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+
+@Data
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+public class Vehicle {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+
+    private String licensePlate;
+
+    private String companyName;
+
+    private String modelName;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JsonBackReference
+    private ParkingTicket parkingTicket;
```

Comment:
Czy tu nie powinno być listy?

Reply:
To takie moje założenie, że dany pojazd ma zawsze tylko jeden bilet z którym jest powiązany co ułatwia znalezienie obecnie używanego biletu, a wiadomo, że nigdy nie jest zaparkowany w dwóch miejscach na raz (nie potrzebny mu drugi bilet w tym samym czasie) i wykorzystane bilety zostają w tabeli bilety, a pojazd dostaje nową encję bilet.

Reply:
To może nazwa zmiennej powinna być bardziej deskryptywna.

Reply:
Hmm, można się nad tym zastanowić - może coś w stylu CurrentlyUsedTicket